File: blog/排序算法整理/bubble-pop/maopao.go
package main

import "fmt"

func main() {
	data := []int{1, 2, 90, 29, 32, 17, 51, 100, 99, 3, 9, 8, 6, 2}
	asc := bubblePop(data, "ASC")
	fmt.Println(asc)
	desc := bubblePop(data, "DESC")
	fmt.Println(desc)
}

func bubblePop(nums []int, order string) []int {
	for i := range nums {
		for j := 0; j < len(nums)-i-1; j++ {
			if nums[j] > nums[j+1] && order == "ASC" || nums[j] < nums[j+1] && order == "DESC" {
				nums[j], nums[j+1] = nums[j+1], nums[j]
			}
		}
	}
	return nums
}

File: blog/排序算法整理/bubble-pop/maopao.js
/**
 * 冒泡排序
 * @param {number[]} nums
 * @param {number[]} order ASC-升序 DESC-降序
 */
function bubblePop (nums, order) {
  const length = nums.length
  for (let i = 0; i < length; i++) {
    for (let j = 0; j < length - i - 1; j++) {
      if (
        (nums[j] > nums[j + 1] && order === 'ASC') ||
        (nums[j] < nums[j + 1] && order === 'DESC')
      ) {
        const temp = nums[j]
        nums[j] = nums[j + 1]
        nums[j + 1] = temp
      }
    }
  }
  return nums
}

const arr = [1, 2, 90, 29, 32, 17, 51, 100, 99, 3, 9, 8, 6, 2]
console.log(bubblePop(arr, 'ASC'))
console.log(bubblePop(arr, 'DESC'))
